MapboxCoreNavigation/RouteProgress.swift

```diff
@@ -388,21 +388,34 @@ open class RouteStepProgress: NSObject {
     /**
      The next intersection the user will travel through.
      
-     The step must contains `Intersections` for this value not be `nil`.
+     The step must contain `Intersections` for this value not be `nil`.
      */
     @objc public var upcomingIntersection: Intersection? {
-        guard let intersections = intersectionsIncludingUpcomingManeuverIntersection, intersectionIndex + 1 < intersections.endIndex else {
+        guard let intersections = intersectionsIncludingUpcomingManeuverIntersection, intersectionIndex + 1 < intersections.endIndex, intersectionIndex >= intersections.startIndex else {
             return nil
         }
         
-        return intersections[intersectionIndex]
+        return intersections[intersections.index(after: intersectionIndex)]
     }
     
     /**
      Index representing the current intersection.
      */
     @objc public var intersectionIndex: Int = 0
     
+    /**
+     The current intersection the user will travel through.
+     
+     The step must contain `Intersections` for this value not be `nil`.
+     */
+    @objc public var currentIntersection: Intersection? {
+        guard let intersections = intersectionsIncludingUpcomingManeuverIntersection, intersectionIndex >= intersections.startIndex, intersectionIndex <= intersections.endIndex else {
```

Comment:
If `intersectionIndex == intersections.endIndex`, then `intersections[intersectionIndex]` will be out of bounds and crash. Consider using the `~=` and `..<` operators.